src/python/pants/goal/context.py

```diff
@@ -155,6 +155,20 @@ def __str__(self):
     ident = Target.identify(self.targets())
     return 'Context(id:{}, targets:{})'.format(ident, self.targets())
 
+  def set_target_root_count_in_runtracker(self):
+    """Sets the target root count in the run tracker's daemon stats object."""
+    # N.B. `self._target_roots` is always an expanded list of `Target` objects as
+    # provided by `GoalRunner`.
+    target_count = len(self._target_roots)
+    self.run_tracker.pantsd_stats.set_target_root_size(target_count)
+    return target_count
+
+  def set_affected_target_count_in_runtracker(self):
+    """Sets the realized target count in the run tracker's daemon stats object."""
+    target_count = len(self.build_graph)
```

Comment:
This is going to break when we stop computing a `BuildGraph` in all cases via #5639 and #4769. And it is explicitly the goal of those tickets to avoid computing this.

@benjyw and I talked about this yesterday, but moving forward, we're going to need to make a decision about what the graph looks like to tasks like `depmap` and `dependencies`... but it's entirely possible that that will not be literally the graph used by a goal like `compile` or `test`, since those will request exactly the products they need in order to execute. This is also touched on in the blog post I sent out about execution models.

I think that a metric that would be more durable as the concept of "targets" evolves might be something like: "number of involved files". There are a few ways to compute that (and ideally it would be computed by the engine itself: see below), but one easy approach for now would be to continue to use the `BuildGraph` (temporarily), and sum the counts of all files owned by targets.

If you have time for the more forwards compatible approach, it would be to add a method similar to [graph_trace](https://github.com/pantsbuild/pants/blob/ac0b7e4b3d8267c682b2e821f8db592ca94103e3/src/rust/engine/src/lib.rs#L434) that counted how many files were accessible below some roots by calling [fs_subject](https://github.com/pantsbuild/pants/blob/ac0b7e4b3d8267c682b2e821f8db592ca94103e3/src/rust/engine/src/nodes.rs#L999): in pseudocode, it would be something like:
```
impl Graph {
  fn count_fs_nodes(&self, roots: Vec<NodeKey>) -> usize {
    self
      .walk(roots.into_iter().flat_map(|n| self.entry_id(EntryKey.Value(n))).collect(), false)
      // Count entries which have an fs_subject.
      .filter(|entry_id| self.entry_for_id.and_then(|e| e.node.content().fs_subject().is_some()))
      .count()
  }
}
```

Reply:
yeah, I fully expect this to evolve over time as we focus on e.g. the runtracker/reporting aspect of the v2 pipeline port. for now, was just going for something basic to establish a better approximate baseline than the accumulating metric we have now for product graph size. I've gone ahead and implemented the file counting idea using `BuildGraph` for the moment w/ a TODO to circle back to do that in the engine.